analyses/compare-gistic/03-GISTIC-gene-level-tally.Rmd

```diff
@@ -0,0 +1,382 @@
+  #                   `focal-cn-file-preparation` or the gene status data.frame 
+  #                   from prepare_gene_level_gistic (set method = "gistic")
+  #   gene_mapping_df: the peak assignment output of prepare_gene_level_gistic 
+  #   method: "gistic" or "exon", see above
+  #   biospecimens_included: a vector of biospecimen identifiers to be used
+  #                          to filter gene_status_df; intended to be used when
+  #                          method = "exon"
+  #   genes_included: a vector of gene symbols to be used to filter 
+  #                   gene_status_df; intended to be used when method = "exon"
+  # 
+  # Returns:
+  #   gene_tally_df: a data.frame with the following columns
+  #                  1. gene: gene symbol
+  #                  2. gain: number of gains tallied
+  #                  3. loss: number of losses tallied
+  #                  4. neutral: number of copy neutral events tallied
+  #                  5. detection_peak: all peaks a gene symbol mapped to 
+  #                                     via GISTIC, comma-separated
+  
+  # TODO: error-handling throughout this function could probably be a bit more
+  # robust
+  if ((method == "exon") & any(c(is.null(biospecimens_included), 
+                                 is.null(genes_included)))) {
+    stop("The 'exon' method is intended to be used with the 
+         biospecimens_included and genes_included arguments")
+  }
+  
+  # If a vector of biospecimen IDs is provided, filter only to those biospecimen
+  # ids
+  if (!is.null(biospecimens_included)) {
+    gene_status_df <- gene_status_df %>%
+      filter(Kids_First_Biospecimen_ID %in% biospecimens_included)
+  }
+  
+  # If a vector of genes is provided, filter only to those genes
+  if (!is.null(genes_included)) {
+    gene_status_df <- gene_status_df %>%
+      filter(gene %in% genes_included)
+  }
+  
+  gene_tally_df <- gene_status_df %>%
+    # Remove any genes that are "nearest genes." We don't expect them to be *in*
+    # the peak and therefore do not expect them to be comparable to the other
+    # method that we used.
+    filter(str_detect(gene, "\\[", negate = TRUE)) %>%
+    # Count the number of times a copy number alteration occurs in an individual
+    # gene
+    group_by(gene, status) %>%
+    tally() %>%
+    spread(status, n) %>%
+    # Any instances where a copy number alteration is NA are equivalent to a 
+    # count of zero
+    replace_na(list(loss = 0, 
+                    gain = 0))
+  
+  # In the GISTIC case, missing values in the neutral column are the same as
+  # zero counts
+  if (method == "gistic") {
+    gene_tally_df <- gene_tally_df %>%
+      replace_na(list(neutral = 0))
+  }
+  
+  # Because neutral calls are not included in the annotated consensus file,
+  # we can assume the number of neutral calls is the number of samples included 
+  # minus any alterations (gain + loss)
+  if (method == "exon") {
```

Comment:
Ah. Maybe just move the `replace_na` step from line 115 to being within this if statement.